pkg/cluster/k8sres.go

```diff
@@ -525,6 +527,52 @@ func (c *Cluster) generatePodTemplate(
 		)
 	}
 
+	monitoringContainerImage := c.OpConfig.MonitoringDockerImage
+	if monitoringDockerImage != nil && *monitoringDockerImage != "" {
+		monitoringContainerImage = *monitoringDockerImage
+	}
+	if monitoringContainerImage != "" {
+		podSpec.Containers = append(
+			podSpec.Containers,
+			v1.Container{
+				Name:            "monitoring-sidecar",
+				Image:           c.monitoringImage(),
+				ImagePullPolicy: v1.PullIfNotPresent,
+				Resources:       *resourceRequirementsMonitoringSidecar,
+				VolumeMounts:    volumeMounts,
+				Env: []v1.EnvVar{
+					{
+						Name: "POD_NAME",
+						ValueFrom: &v1.EnvVarSource{
+							FieldRef: &v1.ObjectFieldSelector{
+								APIVersion: "v1",
+								FieldPath:  "metadata.name",
+							},
+						},
+					},
+					{
+						Name: "POD_NAMESPACE",
+						ValueFrom: &v1.EnvVarSource{
+							FieldRef: &v1.ObjectFieldSelector{
+								APIVersion: "v1",
+								FieldPath:  "metadata.namespace",
+							},
+						},
+					},
+					// TODO: figure out auth.
+					/*
+						{
+							Name:  "POSTGRES_USER",
```

Comment:
You can expose them from the secrets, i.e.
```
ValueFrom: &v1.EnvVarSource{
				SecretKeyRef: &v1.SecretKeySelector{
					LocalObjectReference: v1.LocalObjectReference{
						Name: c.credentialSecretName(c.OpConfig.SuperUsername),
					},
					Key: "password",
				},
			},
```

Reply:
done!